src/process.h

```diff
@@ -43,12 +43,12 @@ struct process {
     uint32_t entry;
     struct fd fd_table[PROCESS_MAX_OPEN_FILES];
     struct process *parent;
-    struct list children;
```

Comment:
have we decided on this parent-children problem? when we keep a children list, when we add the child process to this list, we will add it to the ready queue, which is a side effect of the list module. it's clearly wrong, but there's no easy way to fix it without reworking the list module.